src/ocrd_utils/os.py

```diff
@@ -75,27 +75,38 @@ def unzip_file_to_dir(path_to_zip, output_directory):
     z.close()
 
 @lru_cache()
-def get_ocrd_tool_json(executable):
+def get_ocrd_tool_json(executable, skip_error_print: bool = False):
     """
     Get the ``ocrd-tool`` description of ``executable``.
     """
+    ocrd_tool = {}
     executable_name = Path(executable).name
-    try:
-        ocrd_tool = loads(run([executable, '--dump-json'], stdout=PIPE).stdout)
-    except (JSONDecodeError, OSError) as e:
-        getLogger('ocrd.utils.get_ocrd_tool_json').error(f'{executable} --dump-json produced invalid JSON: {e}')
-        ocrd_tool = {}
+    # TODO: Potentially these executables should be moved as a constant somewhere else
+    skip_executables = ["ocrd-cis-data", "ocrd-import", "ocrd-make"]
+    if executable_name not in skip_executables:
+        try:
+            ocrd_tool = loads(run([executable, '--dump-json'], stdout=PIPE).stdout)
+        except (JSONDecodeError, OSError) as e:
+            if not skip_error_print:
+                getLogger('ocrd.utils.get_ocrd_tool_json').error(f'{executable} --dump-json produced invalid JSON: {e}')
+        except Exception as e:
+            if not skip_error_print:
+                getLogger('ocrd.utils.get_ocrd_tool_json').error(f'{executable} --dump-json failed: {e}')
```

Comment:
IMO it's not a good pattern to suppress these warnings. Either the tool exists – then we _should_ see the error message, or it does not – then we should not even _attempt_ to run it.

The problem in #1218 is that in …
https://github.com/OCR-D/core/blob/79c61e303c87f229d5c96aedc0da31ef82b0f5d3/src/ocrd/resource_manager.py#L42
… the default preinstalled resource database contains 
- outdated processors that we do not use anymore 
- entries that might not in fact be installed currently

So the right thing IMO would be to filter … 
https://github.com/OCR-D/core/blob/79c61e303c87f229d5c96aedc0da31ef82b0f5d3/src/ocrd/resource_manager.py#L85-L87
… by those that can be found in PATH.

Reply:
> entries that might not in fact be installed currently

Why should the resmgr be limited by the fact that a processor is locally installed or not? The models should be downloadable regardless. The resmgr does the ocrd tool json dump by running the processor - the main limitation. Hopefully, that will be resolved with #1250. 

For now, I can still filter these though.

Reply:
You're right – this should be independent. (I think I confused this with the other issue, which covers `list_installed` rather than `list_available`.)

Ok, but perhaps instead of running `--dump-json` and `--dump-module-dir` directly (and having to deal with different cases of error), we should first look whether the executable exists and only _then_ attempt something with a potential error message?

Reply:
> Ok, but perhaps instead of running --dump-json and --dump-module-dir directly (and having to deal with different cases of error), we should first look whether the executable exists

The resource manager inside `list_available` executes the `--dump-json` only for executables found in `PATH`. I think an additional filter is required to make sure the executable name follows the `ocrd-*` pattern. The main issue then is when paths have `ocrd-*` inside as false positives such as `ocrd-make`. These should be either explicitly banned in a list or the `--dump-json` should be executed regardless and the error message should be suppressed (you are not in favor of that option based on your comments above).

Reply:
Ah, indeed, `executable` already is from PATH.

Filtering: yes, we [already agreed false positives should be ruled out in resmgr itself](https://github.com/OCR-D/core/pull/1249/files#r1665531809).

But then – at least for `list-available` I don't see any need to suppress warnings in ocrd_utils.